packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -728,9 +728,12 @@ class _Label extends StatelessWidget {
 
     if (!showUnselectedLabels && !showSelectedLabels) {
       // Never show any labels.
-      text = Opacity(
```

Comment:
Techically these could be left as opacity, and nothing would be hurt. I updated them for consistency, but I could see it either way

Reply:
(here and below) I slightly prefer the old style since it's more compact and doesn't require reasoning about the 
"maintain" properties (see also my other comment about maybe having a separate constructor on Visibility). But no strong feelings.

Reply:
@srawlins for ideas, but I'd be concerned that it would be too difficult to detect "Opacity widget that only has values of 0.0 or 1.0." Technically Opacity widget with values of only 0.0 is fine, and Opacity widget with value of 1.0 is not really common.